spec/rubocop/cop/factory_girl/dynamic_attribute_defined_statically_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::FactoryGirl::DynamicAttributeDefinedStatically do
+  subject(:cop) { described_class.new(config) }
+
+  let(:config) { RuboCop::Config.new }
+
+  it 'registers an offense for offending code' do
+    expect_offense(<<-RUBY)
+      FactoryGirl.define do
+        factory :post do
+          trait :published do
```

Comment:
I would separate the tests for better readability (registers an offense in a trait as an own test case)